docs/docs/recipes/transforming-data.md

```diff
@@ -54,3 +54,65 @@ export const query = graphql`
 
 - [Tutorial on transforming Markdown to HTML](/tutorial/part-six/#transformer-plugins) using `gatsby-transformer-remark`
 - Browse available transformer plugins in the [Gatsby plugin library](/plugins/?=transformer)
+
+## Transforming images into grayscale using GraphQL
+
+### Prerequisites
+
+- A Gatsby site with `gatsby-config.js` and an `index.js` page
+- The `gatsby-transformer-sharp` plugin installed
+- The `gatsby-plugin-sharp` plugin installed
+- A source plugin installed, such as `gatsby-source-filesystem`
+- An image (`.jpg`, `.png`, `.gif`, `.svg`, etc.) in the `src` folder
+
+### Directions
+
+1. Make sure that the config file has the correct information.
+
+```JavaScript:title=gatsby-config.js
+
+ plugins: [
+   {
+     resolve: `gatsby-source-filesystem`,
+     options: {
+       name: `images`,
+       path: `${__dirname}/src/images`,
+     },
+   },
+   `gatsby-transformer-sharp`,
+   `gatsby-plugin-sharp`,
+ ],
+```
+
+Note: Note the gatsby src system path
+
+2.
```

Comment:
What is this step about, and where does this code go?

Even though recipes are concise and short, it still helps to set up a step so the user knows what the code is intended to do, and where they'd put it in their project. To illustrate the example, show a trimmed down React component file so the user can recognize what to do with the query; otherwise, it can be too abstract for people to follow along. There's a `my-dogs.js` page component example on the `Using gatsby-image` guide on step 4 under the first Egghead video, complete with syntax highlighting to draw emphasis on the query: https://www.gatsbyjs.org/docs/using-gatsby-image

Syntax highlighting is outlined in the Style Guide: https://www.gatsbyjs.org/contributing/gatsby-style-guide/#code-formatting-line-highlighting

Do you have a working example for this? You can glean information from that to apply to the recipe and help users follow what you're doing.